05_iigo/index.tex

```diff
@@ -429,51 +452,522 @@ \section{IIGO Session order}
+\end{equation}
+\begin{equation}
+    \textrm{   Auxiliary vector: } 
+    \begin{bmatrix}
+        2 
+    \end{bmatrix}   
+\end{equation}
+However, we can also write both Equation~\eqref{equation:rx1} and Equation~\eqref{equation:rx2} as a single stacked rule as follows:
+\begin{equation}
+    \textrm{Rule matrix: }
+    \begin{bmatrix}
+        0 & 2 & -1 & 0 \\
+        -1 & 1 & 0 & 0 
+    \end{bmatrix}
+\end{equation}
+\begin{equation}
+    \textrm{   Variables to fill:  }
+    \begin{bmatrix}
+        \textrm{Amount of tax agent is asked to pay} \\
+        \textrm{Amount of tax the agent is actually paying} \\
+        \textrm{Amount of resources agent has} \\
+        1
+    \end{bmatrix}     
+\end{equation}
+\begin{equation}
+    \textrm{   Auxiliary vector: } 
+    \begin{bmatrix}
+        2 \\
+        2 
+    \end{bmatrix}   
+\end{equation}
+The advantage of this matrix-based rule representation is that the mechanics of the calculation is exposed. 
+Any agent can inspect the elements of the rule matrix in order to easily calculate what values they must provide to adhere to a specific rule.
+For example, if an agent is looking at the above rule and knows that their resource count is $120$, and also know they've been asked to pay $75$, they can go line by line through the matrix to work our what the minimum amount of tax that is needed to be paid.
+\begin{equation}
+    \textrm{Matrix line 1: }
+    \begin{bmatrix}
+        0 & 2 & -1 & 0
+    \end{bmatrix}
+\end{equation}
+\begin{equation}
+    \textrm{Values to be fed in: }
+    \begin{bmatrix}
+        75 \\
+        x \\
+        120 \\
+        1
+    \end{bmatrix}
+\end{equation}
+\begin{equation}
+    \textrm{Agent calculation: }
+    0*75 + 2x -120 \geq 0
+\end{equation}
+\begin{equation}
+    x \geq 60
+\end{equation}
+A similar calculation for the next line of the matrix yields:
+\begin{equation}
+    \textrm{Agent calculation: }
+    x \geq 75
+\end{equation}
+Following the equations above, the agent is able to understand that it needs to pay a minimum of $75$ for tax. This system allows the agents both to check if they are compliant and if not, how to calculate which values they need to be compliant.
+As a final note, we have designed an additional abstraction to this rules system to accomodate conditional rules.
+For example, if you had the rule:
+\begin{rule_IIGO}
+    "If resources are above $100$, you must pay at least $75$".
+\end{rule_IIGO} 
+To accommodate this, we instead split this into two separate rules.
+\begin{rule_IIGO}
+    "If resources are above $100$,"
+    \label{rule:cond1}
+\end{rule_IIGO}
+\begin{rule_IIGO}
+    "You must pay at least $75$."
+    \label{rule:cond2}
+\end{rule_IIGO}
+We then conditionally execute them such that if Rule~\ref{rule:cond1} fails then the whole structure passes, and if Rule~\ref{rule:cond1} passes, we expect Rule~\ref{rule:cond2} to pass as well for the rule to be adhered.
+\subsection{Rules In Play}
```

Comment:
rogue subsection

Reply:
this is TODO.. that is why it is there :)

Reply:
but yea, we can delete it or smth